doc/making-a-visualizer.md

```diff
@@ -1,17 +1,212 @@
 ## Building a Visualizer
 
-Big picture, every visualizer needs to implement the interface defined in
-`VisualizerInterface.ts` which specifies the basic expectations of a
-visualizer. (See below for the easy way to do this.) This interface includes
-the following data and methods:
+### Get started
+
+#### Choose a graphics framework
+
+The easiest way to build a visualizer is to extend a pre-made visualizer base
+class, which automatically sets up a graphics framework for you to use. Right
+now, there's only one base class available:
+
+-   [`P5Visualizer`](#p5-visualizers) uses the
+    [**p5.js**](https://p5js.org/learn/) library for graphics and user
+    interaction.
+
+If you want to use a new graphics framework, you'll need to write your own
+implementation of the [visualizer interface](#abstract-visualizers).
+
+#### Test your visualizer on the workbench
+
+While you're working on a visualizer, we recommend keeping it in the
+`src/visualizers-workbench` directory, where Frontscope typically won't notice
+it. To load it, run Frontscope in "workbench mode" by calling
+`npm run dev:workbench`.
+
+When you're ready to propose your visualizer as an official part of
+Numberscope, you'll move it to `src/visualizers`. Then Frontscope will notice
+and load it when you call the usual `npm run dev`. For details, see
+[below](#where-to-put-your-visualizer).
+
+#### Document your visualizer
+
+Each visualizer has its documentation woven into the source code, using
+special `/** md`&nbsp;…&nbsp;`**/` comments that are automatically compiled
+into a
+[documentation page](https://numberscope.colorado.edu/doc/src/visualizers/Differences/).
```

Comment:
You are using an external reference to the production doc page for Differences; why not an internal reference `(../src/visualizers/Differences.md)`?

Reply:
I didn't realize that would work both on the doc pages and on GitHub! Done.